service/internal/builder/exporters_builder_test.go

```diff
@@ -89,7 +90,7 @@ func TestBuildExporters(t *testing.T) {
 	// This should result in creating an exporter that has none of consumption
 	// functions set.
 	delete(cfg.Service.Pipelines, "trace")
-	exporters, err = BuildExporters(zap.NewNop(), trace.NewNoopTracerProvider(), component.DefaultBuildInfo(), cfg, factories.Exporters)
+	exporters, err = BuildExporters(zap.NewNop(), trace.NewNoopTracerProvider(), metric.NoopMeterProvider{}, component.DefaultBuildInfo(), cfg, factories.Exporters)
```

Comment:
Pass TelemetrySettings in the builder

Reply:
Updated this, please take a look.

Reply:
Can you remove the changes in config for the moment? We don't do the same for trace.

Reply:
We can have a follow-up if needed

Reply:
Do you mean split the PR into one to guard against invalid settings and another to pass in the TelemetrySettings from the service?